lib/cusparse/linalg.jl

```diff
@@ -0,0 +1,35 @@
+using LinearAlgebra
+
+function sum_dim1(A::CuSparseMatrixCSR)
+    function kernel(Tnorm, out, dA)
+        idx = (blockIdx().x-1) * blockDim().x + threadIdx().x
+        idx < length(dA.rowPtr) || return
+        s = zero(Tnorm)
+        for k in dA.rowPtr[idx]:dA.rowPtr[idx+1]-1
+            s += abs(dA.nzVal[k])
+        end
+        out[idx] = s
+        return
+    end
+
+    m, n = size(A)
+    Tnorm = typeof(float(real(zero(eltype(A)))))
+    Tsum = promote_type(Float64,Tnorm)
+    rowsum = CUDA.CuArray{Tsum}(undef, m)
+    kernel_f = @cuda launch=false kernel(Tnorm, rowsum, A)
+    
+    threads = min(n, CUDA.maxthreads(kernel_f))
```

Comment:
Why not use the occupancy API as we normally do?

Reply:
I won't be able to query threads if I don't have the kernel function object?

Reply:
Look for `launch_configuration` in the source code, it also gives you the max threads by using the kernel object.